### PR TITLE
Refacto perf

### LINKED ISSUE #301: Performance improvement
We really need to discuss that. Since we started using BABEL_DISABLE_CACHE, everything is super slow.

First, using BABEL_DISABLE_CACHE for the static build is brutal: we should instead just "remove" the cache before doing the build, since during the build we are not modifying the files and can benefit from the cache, this should speed up things a lot (I hope).

Second, during development, maybe we should consider trying (again) a more smart cache cleaner that will remove (eg: when a css file change) the appropriate cache _only_ (not everything).
Maybe we should handle that in babel-plugin-webpack-loaders (poke @istarkov).

Any thoughts?

### PR DESCRIPTION
**Pull request by MoOx**
Best to review with [?w=1](https://github.com/MoOx/phenomic/pull/422/files?w=1)

Closes #301 

Since there is a pretty considerable change in the `scripts`, that's probably the time to handle those issue before cutting a release:

- generate all phenomic requirements #412
- remove redux and the store #40
- validate CLI args #363

### CODE REVIEW

**Comment by thangngoc89** [comment]
:+1: Looks fantastic on a quick run. I'll take a closer look tonight

**Comment by MoOx** [comment]
The issue on Appveyor seems related to a babel-register issue. Will take a look later.

File: docs/content/docs/faq.md [hunk 1]
```diff
@@ -2,6 +2,20 @@
...
 
+## ``[BABEL] Note: The code generator has deoptimised the styling of ...``
+
+_This is not an issue. Just a simple information._
+
+When babel compiles some code, it try to offer a readable output, but when files
+become big (>100KB), babel considers (by default) that it's not useful to keep
+a readable output.
+If you are annoyed by this message, or if you don't care about readability,
+you can just use `compact: true`.
```
Comment (thangngoc89) [inline docs/content/docs/faq.md]: Could you make this default option ?

File: docs/scripts/phenomic.node.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import metadata from "../web_modules/app/metadata"
+import routes from "../web_modules/app/routes"
+import store from "../web_modules/app/store"
+
+import phenomicStatic from "phenomic/lib/static"
+
+module.exports = (options) =>
```
Comment (thangngoc89) [inline docs/scripts/phenomic.node.js]: Why not `export default` here ?

File: src/builder/cleanup.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { join } from "path"
+import { unlinkSync } from "fs"
+
+export default (wd, fileToKeep, assets) => {
+  assets
+    .filter((file) => file !== fileToKeep)
+    .forEach((file) => {
+      unlinkSync(join(wd, file))
```
Comment (thangngoc89) [inline src/builder/cleanup.js]: How about using `unlink` promise and `Promise.all` ? I think it would help the perf

File: src/bin/index.js [hunk 4]
```diff
@@ -1,8 +1,18 @@
 #!/usr/bin/env node
 
+if (!process.env.DEBUG) {
+  process.env.DEBUG = "phenomic:*"
+}
+
 // still in src/, handy to play
 if (__filename.indexOf("phenomic/src") > 1) {
```
Comment (thangngoc89) [inline src/bin/index.js]: I think this is why build failed on Appveyor. This requires the phenomic source folder must be named `phenomic`. Maybe you can try with `indexOf("src/bin/index.js")`

File: docs/scripts/phenomic.node.js (hunk 2 above)
Comment (MoOx) [inline docs/scripts/phenomic.node.js]: node compat. This script is consumed via node. But I will try to use export default and adjust in the core usage

File: docs/content/docs/faq.md (hunk 1 above)
Comment (MoOx) [inline docs/content/docs/faq.md]: Maybe. But I don't think "true" or "false" are better. Auto is a good choice. But why not forcing true by default.

File: src/builder/cleanup.js (hunk 3 above)
Comment (MoOx) [inline src/builder/cleanup.js]: I am lazy, feel free to adjust this ;)

File: src/bin/index.js (hunk 4 above)
Comment (MoOx) [inline src/bin/index.js]: This is unchanged, but maybe that's the reason since I changed something in another place. Will try something else.

Comment (thangngoc89) [inline src/bin/index.js]: Yup. I noticed that. But I didn't change my local folder (stills statinamic`) and it throws the extract error on Appveyor when I build docs.

File: docs/content/docs/faq.md (hunk 1 above)
Comment (thangngoc89) [inline docs/content/docs/faq.md]: yeah. User can always change this. But no warnings by default is better

File: src/builder/cleanup.js (hunk 3 above)
Comment (thangngoc89) [inline src/builder/cleanup.js]: I'll leave it there for now. This will be a lot easier if we use async/await for `src/builder/index.js`

Comment (thangngoc89) [inline src/builder/cleanup.js]: https://ci.appveyor.com/project/MoOx/phenomic/build/1580#L2452

Path on windows is different from linux. ('/' vs '\')

File: src/bin/index.js (hunk 4 above)
Comment (thangngoc89) [inline src/bin/index.js]: https://ci.appveyor.com/project/MoOx/phenomic/build/1580#L2452

Path on windows is different from linux. ('/' vs '\')

**Comment by thangngoc89** [comment]
I fixed Windows compat issue.

File: src/bin/index.js [hunk 5]
```diff
@@ -5,7 +5,10 @@ if (!process.env.DEBUG) {
 }
 
 // still in src/, handy to play
-if (__filename.indexOf("phenomic/src") > 1) {
+if (
+  __filename.endsWith("src/bin/index.js") ||
+  __filename.endsWith("src\\bin\\index.js")
```
Comment (MoOx) [inline src/bin/index.js]: can you use node path.sep instead ? Or even better, join("src", "bin", "index.js").

Comment (thangngoc89) [inline src/bin/index.js]: I updated it with `path.join`

**Comment by thangngoc89** [comment]
LGTM

**Comment by kuzvac** [comment]
If you remove redux/store we can use it like regular redux/store or how?

**Comment by thangngoc89** [comment]
@kuzvac you can add it back like you would do in any React project.

**Comment by MoOx** [comment]
Yeah sure, that's planned ;)

**Comment by kuzvac** [comment]
Ok, thank you all :)

**Pull request merged**